lms/static/scripts/frontend_apps/components/test/BookPicker-test.js

```diff
@@ -4,30 +4,37 @@ import { act } from 'preact/test-utils';
 
 import mockImportedComponents from '../../../test-util/mock-imported-components';
 import { waitForElement } from '../../../test-util/wait';
+import { Services } from '../../services';
 import * as bookData from '../../utils/vitalsource-sample-data';
 import BookPicker, { $imports } from '../BookPicker';
 
 describe('BookPicker', () => {
-  let fakeAPI;
+  let fakeVitalSourceService;
+
+  const BookPickerWrapper = props => {
+    const services = { vitalSource: fakeVitalSourceService };
+    return (
+      <Services.Provider value={services}>
+        <BookPicker {...props} />
+      </Services.Provider>
+    );
+  };
```

Comment:
This wrapper is needed for tests that want to re-render the component using different props with `wrapper.setProps(...)`. This wouldn't work if the JSX tree here were passed to directly to `mount`, since the root component is now a `Services.Provider` which accepts only a `value` prop.

Reply:
I wonder if we could write a test utility/wrapper for this...

Reply:
Done. There is now a `withServices` function defined in `services/index.js` for this. Even though it is primarily intended for use in tests, I defined it alongside `useServices` just so all the parts are together.